ui/src/pages/Dashboard/Error.tsx

```diff
@@ -2,26 +2,163 @@ import React, { useEffect } from 'react';
 import { DashboardCore } from '@splunk/dashboard-core';
 import { DashboardContextProvider } from '@splunk/dashboard-context';
 import EnterpriseViewOnlyPreset from '@splunk/dashboard-presets/EnterpriseViewOnlyPreset';
+import Button from '@splunk/react-ui/Button';
+import { variables } from '@splunk/themes';
+import styled from 'styled-components';
+import SearchJob from '@splunk/search-job';
+
 import { getActionButtons, waitForElementToDisplayAndMoveThemToCanvas } from './utils';
+import { OPEN_SEARCH_LABEL } from './ErrorPageConfig';
+import { getUnifiedConfigs } from '../../util/util';
+
+const OpenSearchStyledBtn = styled(Button)`
+    max-width: fit-content;
+    font-size: ${variables.fontSize};
+    align-content: flex-end;
+    place-self: end;
+    grid-column: 2;
+    grid-row: 4;
+    bottom: 8px;
+`;
+
+// const OpenTroubleshootingBtn = styled(OpenSearchStyledBtn)` // commented out for now but will be needed latter on
+//     margin-right: 145px;
+// `;
+
+let apiReference: { updateDefinition: (arg0: Record<string, unknown>) => void } | null = null;
```

Comment:
the state out of react lifecycle seems scary

Reply:
i tried before to keep it inside component but for some reason there is error when using useRef, i will try one more time, maybe i made some mistake earlier.
Also if i wont find reason why it does not work i should probably assign null when component unmounts. (forgot about that one)

Reply:
not sure why it did not work last time (maybe i did something wrong but right now i used useRef and works like a charm)